src/extension_commands.ml

```diff
@@ -534,6 +534,143 @@ module Copy_type_under_cursor = struct
     command Extension_consts.Commands.copy_type_under_cursor handler
 end
 
+module Search_by_type = struct
+  let extension_name = "Search By Type"
+
+  let ocaml_lsp_doesnt_support_search_by_type instance ocaml_lsp =
+    match
+      Ocaml_lsp.is_version_up_to_date
+        ocaml_lsp
+        (Extension_instance.ocaml_version_exn instance)
+    with
+    | Ok () -> ()
+    | Error (`Msg msg) ->
+      show_message
+        `Warn
+        "The installed version of `ocamllsp` does not support type search. %s"
+        msg
+
+  let rec remove_duplicates = function
+    | (a : Custom_requests.Type_search.type_search_result) :: (b :: _ as t) ->
+      if String.equal a.name b.name && a.cost = b.cost then remove_duplicates t
+      else a :: remove_duplicates t
+    | unique -> unique
+
+  let get_search_results ~query ~limit ~with_doc ~position text_editor client =
+    let open Promise.Syntax in
+    let doc = TextEditor.document text_editor in
+    let uri = TextDocument.uri doc in
+    let* res =
+      Custom_requests.(
+        send_request
+          client
+          Type_search.request
+          (Type_search.make ~uri ~position ~limit ~query ~with_doc ()))
+    in
+    match res with
+    | [] ->
+      show_message `Info "Empty results";
+      [] |> Promise.return
+    | res -> remove_duplicates res |> Promise.return
+
+  let display_search_results results =
+    Vscode.(
+      Window.showQuickPickItems
+        ~choices:
+          (List.map
+             results
+             ~f:(fun (res : Custom_requests.Type_search.type_search_result) ->
+               ( Vscode.QuickPickItem.create
+                   ~label:res.name
+                   ~description:res.typ
+                   ~detail:(Option.value ~default:"" res.doc)
+                   ()
+               , res.name )))
+        ~options:
+          (QuickPickOptions.create
+             ~title:"Type/Polarity Search Results"
+             ~canPickMany:false
+             ~ignoreFocusOut:true
+             ())
+        ())
+
+  let _search_by_type =
+    let handler (instance : Extension_instance.t) ~args:_ =
+      let search_by_type () =
+        match Window.activeTextEditor () with
+        | None ->
+          Extension_consts.Command_errors.text_editor_must_be_active
+            extension_name
+            ~expl:"The command to search for a value by it's type"
```

Comment:
Looking at the behavior of `text_editor_must_be_active`, the current `~expl`anation doesn't seem adapted:
```ocaml
module Command_errors = struct
  let text_editor_must_be_active ~expl cmd_name =
    Printf.sprintf
      "The command \"OCaml: %s\" should be run only with a file open in the \
       editor. %s"
      cmd_name
      expl
end
```

Reply:
The correct explanation is that the command needs the cursor position to deduce the environment and know were to insert the value.